modules/gateway/nodes.go

```diff
@@ -0,0 +1,70 @@
+package gateway
+
+import (
+	"errors"
+	"math/rand"
+
+	"github.com/NebulousLabs/Sia/encoding"
+	"github.com/NebulousLabs/Sia/modules"
+)
+
+const (
+	maxSharedNodes = 10
+	maxAddrLength  = 100
+	minPeers       = 3
+)
+
+// addNode adds an address to the set of nodes on the network.
+func (g *Gateway) addNode(addr modules.NetAddress) error {
+	if _, exists := g.nodes[addr]; exists {
+		return errors.New("node already added")
+	}
+	g.nodes[addr] = struct{}{}
+	g.log.Println("INFO: added node", addr)
+	return nil
+}
```

Comment:
perhaps, though it would make more sense to do it in `Connect` rather than `addNode`.
currently the only time we request nodes is in `Bootstrap`. Eventually we should add ongoing advertisements of our own address, e.g. every 24 hours.